lang/cs/Org.Apache.REEF.IMRU/API/UpdateResult.cs

```diff
@@ -50,22 +50,34 @@ internal bool IsNotDone
             get { return !_done; }
         }
 
-        internal bool HasMapInput
+        /// <summary>
+        /// Indicates if the UpdateResult has map input data
+        /// </summary>
+        public bool HasMapInput
```

Comment:
> Since UpdateResult is exposed to the IMRU example as public, its property can be public as well. 

This is false in the general case. Just because a type is `public` doesn't mean all its members have to be public. It also is generally advisable to hide implementation details from users of a type.

In the case here, `UpdateResult` is used merely as a container to hold on to the results of the update function. Its `internal` methods were introduced only for the internal logic of IMRU. Hiding these members from the caller allow us to change that logic in the future, e.g. make changes to `UpdateResult` that remove these members. Making them `public` prevents us from doing that and constrains future evolution of our APIs.

> UpdateResult is defined in IMRU and used in IMRU.Example in task functions

I assume by "task functions" you mean `BroadcastSenderReduceReceiverUpdateFunctionFT`, right? I don't think this warrants the addition of a new `public` API. `BroadcastSenderReduceReceiverUpdateFunctionFT` uses `UpdateResult` as the representation of its state. This is why it needs access to the members of `UpdateResult`.

However, this situation seems highly unlikely in real applications, which are certain to have their own internal state representation. For instance, a simple ML algorithm might have the last model and current learning rate as its state, while only the model was broadcast from the `UpdateFunction` to the workers.

Hence, I suggest we change the example to have internal state different from `UpdateResult` and leave these members `internal` to allow us flexibility down the line.